dummy_ticket.mligo

```diff
@@ -0,0 +1,33 @@
+type storage = int * bytes ticket list
+
+type return = operation list * storage
+
+type vault_ticket = bytes ticket
+type vault_deposit = {
+  ticket: vault_ticket;
+  address: address
+}
+
+type parameter =
+| Deposit of {
+  deku_consensus : address;
+  deku_recipient : address
+}
+| Withdraw of bytes ticket
+
+let main (param, (counter, stored_tickets) : parameter * storage) : return =
+ match param with
+ | Deposit {deku_consensus; deku_recipient} ->
+   (match (Tezos.get_entrypoint_opt "%deposit" deku_consensus : vault_deposit contract option)  with 
+   | Some contract ->
+    (* Content of ticket is just a packed int *)
+    let ticket = Tezos.create_ticket (Bytes.pack counter) 1n in 
+    let param  = { ticket = ticket; address = deku_recipient } in
+    let op = Tezos.transaction param 0mutez contract in
+    (* We increment the counter to add some variety to the data *)
+    [op], (counter + 1, stored_tickets)
+   | None -> failwith "Entrypoint doesn't exist")
+  | Withdraw ticket ->
+    let ops : operation list = [] in
+    ops , (counter, ticket :: stored_tickets)
```

Comment:
Just to be sure: calling this contract's "deposit" entrypoint amounts to creating a ticket belonging to `deku_recipient`, who will ultimately be the dummy_ticket contract, and sending it to the Deku vault contract. This ticket only contains the current counter value. Correct?

Withdrawing a ticket should only be done through `deku_cli withdraw`. Right?

Reply:
Not quite.

The `Deposit` entrypoint of dummy ticket contract forges a ticket, and then calls the consensus contract's `deposit` entrypoint with the parameters. ` {ticket = ticket; address = deku_recipient}`. At that point, you'll have to read `consensus.mligo` to see what it does with it, but I can summarize it by saying it locks the ticket in a vault and tags it as being owned by the `address` specified. When the deku nodes hear this transaction, they forge the ticket on the Deku side of things, and the address originally listed in `deku_recipient` can now use the asset. 

Withdraw is more complex, please see https://hackmd.io/L74tCg4cTjq9AO7JIQq2BA